handler.go

```diff
@@ -46,13 +46,18 @@ func (g *gzipHandler) Handle(c *gin.Context) {
 	}
 
 	gz := g.gzPool.Get().(*gzip.Writer)
-	defer g.gzPool.Put(gz)
-	defer gz.Reset(ioutil.Discard)
 	gz.Reset(c.Writer)
 
 	c.Header("Content-Encoding", "gzip")
 	c.Header("Vary", "Accept-Encoding")
-	c.Writer = &gzipWriter{c.Writer, gz}
+	c.Writer = &gzipWriter{
+		ResponseWriter: c.Writer, writer: gz,
+		onFlush: func() {
+			gz.Reset(ioutil.Discard)
```

Comment:
interesting idea. So that mean that we will still use this writer on error right? But in opposite to what we have now - it would not be moved back to pool at that time and everything should work then?

Lets try